packages/shell/src/views/HeaderView.ts

```diff
@@ -150,6 +174,17 @@ export class XHeaderView extends BaseView {
     this.command({ type: "clear-authentication" });
   }
 
+  private handleToggleClick(e: Event) {
+    e.preventDefault();
+    e.stopPropagation();
+    this.dispatchEvent(
+      new CustomEvent("toggle-view", {
```

Comment:
`toggle-view` is pretty generic (we have a lot of views!) -- maybe be more specific e.g. `toggle-charm-view`

though we also have a lot of "charm views" as well! Maybe the top level state could be "showDefaultCharm" (the inverse being showing the "native charm list") -- or instead of flipping, could be "showNativeCharmList" (slightly more unique)